tests/unit/backend/corpora/api_server/curator/collection/test_curation_collections.py

```diff
@@ -409,16 +410,118 @@ def test__update_collection__OK(self):
+            publisher_metadata=publisher_metadata,
+        ).id
+        new_name = "A new name, and only a new name"
+        metadata = {"name": new_name}
         response = self.app.patch(
             f"/curation/v1/collections/{collection_id}",
             data=json.dumps(metadata),
             headers=self.make_owner_header(),
         )
         self.assertEqual(200, response.status_code)
         response = self.app.get(f"curation/v1/collections/{collection_id}")
-        self.assertEqual(response.json["name"], metadata["name"])
+        self.assertEqual(new_name, response.json["name"])
+        self.assertEqual(description, response.json["description"])
+        self.assertEqual(contact_name, response.json["contact_name"])
+        self.assertEqual(contact_email, response.json["contact_email"])
+        self.assertEqual(links, response.json["links"])
+        self.assertEqual(publisher_metadata, response.json["publisher_metadata"])
+
+    def test__update_collection__links_and_doi_management__OK(self):
+        name = "partial updates test collection"
+        links = [
+            {"link_name": "name", "link_type": "RAW_DATA", "link_url": "http://test_link.place"},
+            {"link_name": "doi", "link_type": "DOI", "link_url": "http://doi.doi"},
+        ]
+        new_links = [
+            {"link_name": "new link", "link_type": "RAW_DATA", "link_url": "http://brand_new_link.place"},
+            {"link_name": "new doi", "link_type": "DOI", "link_url": "http://doi.org/10.1016"},  # a real DOI
+        ]
+        new_links_no_doi = [
+            {"link_name": "new link", "link_type": "RAW_DATA", "link_url": "http://brand_new_link.place"},
+        ]
+
+        links_configurations = (
+            ("With links already in place; new links replace old", links, new_links, 200, new_links),
+            ("With no links in place; new links get added", None, new_links, 200, new_links),
+            ("With links in place, but empty request; no changes are made", links, None, 200, links),
+            ("With links in place, empty array passed; BAD REQUEST 400", links, [], 400, links),
+            ("With links but no doi; doi remains in place", links, new_links_no_doi, 200, new_links_no_doi + links[1:]),
+        )
+
+        for test_title, initial_links, new_links, expected_status_code, expected_links in links_configurations:
+            with self.subTest(test_title):
+                collection_id = self.generate_collection(self.session, name=name, links=initial_links).id
+                original_collection = self.app.get(f"curation/v1/collections/{collection_id}").json
+                self.assertEqual(initial_links if initial_links else [], original_collection["links"])
+                metadata = {"links": new_links} if new_links is not None else {}
+                response = self.app.patch(
+                    f"/curation/v1/collections/{collection_id}",
+                    data=json.dumps(metadata),
```

Comment:
```suggestion
                    json=metadata,
```

Reply:
that doesn't work. It requires JSON.

Reply:
it does work https://github.com/chanzuckerberg/single-cell-data-portal/blob/d5423ef0bdb75f8b345a302d16875f2a42f5dfbc/tests/unit/backend/corpora/api_server/curator/collection/test_dataset.py#L69

Reply:
Ah, sorry, I tried it with `data=`.